src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_bug_trio.cpp

```diff
@@ -47,6 +47,28 @@ enum Events
     EVENT_YAUJ_CHECK      = 3
 };
 
+struct boss_bug_trioAI : public ScriptedAI
+{
+public:
+    boss_bug_trioAI(Creature* creature) : ScriptedAI(creature) { instance = me->GetInstanceScript(); }
+
+    void AggroTrio(Unit* pUnit)
+    {
+        if (Creature* vem = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_VEM)))
+            if (vem->GetGUID() != me->GetGUID())
+                vem->GetAI()->AttackStart(pUnit);
+        if (Creature* kri = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_KRI)))
+            if (kri->GetGUID() != me->GetGUID())
+                kri->GetAI()->AttackStart(pUnit);
+        if (Creature* yauj = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_YAUJ)))
+            if (yauj->GetGUID() != me->GetGUID())
+                yauj->GetAI()->AttackStart(pUnit);
```

Comment:
little suggestion: you could merge the conditions into **one if** here:
```C++
        if (Creature* vem = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_VEM)) && vem->GetGUID() != me->GetGUID())
            vem->GetAI()->AttackStart(pUnit);
        if (Creature* kri = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_KRI)) && kri->GetGUID() != me->GetGUID())
            kri->GetAI()->AttackStart(pUnit);
        if (Creature* yauj = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_YAUJ)) && yauj->GetGUID() != me->GetGUID())
            yauj->GetAI()->AttackStart(pUnit);
```

Reply:
That's true, will do. Thanks for the suggestion!, greatly appreciated :D

Reply:
Aha, I remember why I did it the way it currently is. I tried something similar to the above when originally writing it but it gave a parsing error.